src/routes/root/Summary.tsx

```diff
@@ -4,32 +4,69 @@ import { useState } from 'react';
 import { Heading } from '../../components/Heading';
 import { Button } from '../../components/Button';
 import { useEnsuredWallet } from '../../context/WalletContext';
-import { DemoAbi as Abi } from '../../abi';
 import { Networks, getNetwork } from '../../utils/network';
+import { DepositAdapterABI, DEPOSIT_ADAPTER_ORIGIN } from '../../contracts';
+import { DepositDataJSON } from '../../components/JSONDropzone/validation';
+import { useStorage } from '../../context/StorageContext';
 
 export function Summary(): JSX.Element {
   const wallet = useEnsuredWallet();
   const [network, setNetwork] = useState<Networks | null>(null);
   const [errorMsg, seteErrorMsg] = useState<string>();
 
+  const storage = useStorage();
+
   // mocks
   const value = '{value}';
   const currency = '{currency}';
 
   const navigate = useNavigate();
-  const handleBridgeClick = (): void => {
-    console.log('do magic on click');
+
+  //TODO - improve code
+  const handleBridgeClick = async (): Promise<void> => {
+    console.log(!network || !storage.data.json);
+    if (!network || !storage.data.json) return;
+
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const contract: Contract<typeof Abi> = new wallet.web3.eth.Contract(
-      Abi,
-      '0x9d15e18Aed0568FB829b857BA1acd1ac8fd68474',
+    const accounts = await wallet.web3.eth.getAccounts();
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const depositAdapterContract: Contract<typeof DepositAdapterABI> =
+      new wallet.web3.eth.Contract(DepositAdapterABI, DEPOSIT_ADAPTER_ORIGIN, {
+        from: accounts[0],
```

Comment:
This needs to be better handled:
- pretty sure we need to listen to accounts changing:
```
const connection = await web3Modal.connect()

// Subscribe to accounts change
connection.on("accountsChanged", (accounts) => {
  console.log('ACCOUNTS CHANGED' + accounts[0]);
});
```
- we need to listen on chain changing
```
provider.on("chainChanged", (chainId: number) => {
console.log(chainId);
});
```
- we need to display current account address and chain

Reply:
is it a more straightforward way to use the RPC provider for listening events?
Changing accounts and forcing users to be on the "right" one can bring a lot of logical issues

Reply:
I think changing account doesn't affect anything really in this case.
If user wants to call deposit method from another account nothing really changes, or does it?
UI will show address changed.

Changing network could introduce some confusion, so I handled it with checking if selected network is the same as the one from provider.

Reply:
> I don't think you can use raw rpc as it won't work with walletconnect. @irubido What happens if user changes account or chain in metamask and you call this method? Will it get invoked with old account that initially connected to website?

I checked it, it will call method from the newly switched address, metamask promps user again for connect if address is switch in extension after connect.